src/store/ducks/opensrp/assignments/index.ts

```diff
@@ -91,15 +92,41 @@ export const removeAssignmentsAction: RemoveAssignmentsAction = {
  * @returns {FetchAssignmentsAction} - action with assignments payload that is added to store
  */
 export const fetchAssignments = (assignmentsList: Assignment[]): FetchAssignmentsAction => {
-  const assignmentsByPlanId: AssignmentsByPlanId = {};
+  // const assignmentsByPlanId: AssignmentsByPlanId = {};
+  const defaultArrayAssignmentHandler: ProxyHandler<object> = {
+    /**
+     * The methods inside the handler object are called traps for our case we use a get trap
+     * @param {AssignmentsByPlanId} target
+     * @param {string} plan planId the assignmentsByPlanId will be updated by empty arrays keyed by the planId
+     */
+    get: (target: AssignmentsByPlanId, plan: string) => {
+      /**
+       * Check if the planId already exists then return the target else add the key with an empty array value
+       */
+      if (plan in target) {
+        return target;
+      } else {
+        target[plan] = [];
+        return target;
```

Comment:
I think the first `return` negates the need for the `else` 😄

Reply:
@cKellyDesign I think the return will only be called when plan exist in the target otherwise we add the plan as a key and assign an empty array.

Reply:
yup totally, my nit pick was about syntax:
```ts
if (plan in target) {
  return target;
} // no else required

target[plan] = [];
return target;
```